core/dbt/jsonschemas.py

```diff
@@ -61,13 +61,25 @@ def jsonschema_validate(schema: Dict[str, Any], json: Dict[str, Any], file_path:
     errors: Iterator[ValidationError] = validator.iter_errors(json)  # get all validation errors
 
     for error in errors:
+
         if error.validator == "additionalProperties" and len(error.path) == 0:
             key = re.search(r"'\S+'", error.message)
             deprecations.warn(
                 "custom-top-level-key-deprecation",
                 msg="Unexpected top-level key" + (" " + key.group() if key else ""),
                 file=file_path,
             )
+        elif (
+            error.validator == "anyOf"
+            and (list(error.path))[-1] == "config"
+            and isinstance(error.instance, dict)
+        ):
+            deprecations.warn(
+                "custom-key-in-config-deprecation",
+                key=(list(error.instance.keys()))[0],
```

Comment:
Similar to above, could error.instance.keys be empty?

Reply:
In this case, `error.instance.keys` will never be empty. Instance is the thing that was validated. If it was nothing, then there would have been nothing to validate, and the failure wouldn't of happened. I can add a guard just in case though anyway.